src/xpi.js

```diff
@@ -0,0 +1,80 @@
+import yauzl from 'yauzl';
+
+
+export default class Xpi {
+
+  constructor(filename, zipLib=yauzl) {
+    this.filename = filename;
+    this.zipLib = zipLib;
+    this.metadata = null;
+    this.zipfile = null;
+  }
+
+  handleEntry(entry, reject) {
+    if (/\/$/.test(entry.fileName)) {
+      return;
+    }
+    // Collect metadata for files.
+    this.zipfile.openReadStream(entry, (err) => {
+      if (err) {
+        return reject(err);
+      }
+      this.metadata[entry.fileName] = entry;
+    });
+  }
+
+  getMetaData(onEventsSubscribed) {
+    return new Promise((resolve, reject) => {
+      // If we have already processed the file and have data
+      // on this instance return that.
+      if (this.metadata && this.zipfile) {
+        return resolve(this.metadata);
+      }
+
+      this.metadata = {};
+      this.zipLib.open(this.filename, {autoClose: false},
+                       (err, zipfile) => {
+        if (err) {
+          reject(err);
+        }
+        this.zipfile = zipfile;
+        zipfile.on('entry', (entry) => {
+          this.handleEntry(entry, reject);
+        });
+        // When the last entry has been processed
+        // cache the metadata and resolve the promise.
+        zipfile.on('end', () => resolve(this.metadata));
+        if (onEventsSubscribed) {
+          // Run optional callback when we know the event handlers
+          // have been inited. Useful for testing.
+          if (typeof onEventsSubscribed === 'function') {
+            onEventsSubscribed();
+          }
+        }
+      });
+    });
+  }
+
+  getFileAsStream(path) {
+    return new Promise((resolve, reject) => {
+      if (!this.zipfile) {
+        throw new Error('this.zipfile is null. Make sure to ' +
+                        'run getMetaData first');
+      }
+
+      if (Object.keys(this.metadata).indexOf(path) === -1) {
+        throw new Error('path does not exist in metadata', path);
+      }
+
+      var entry = this.metadata[path];
```

Comment:
`entry` is only used once on the very next line. Is this here to cut down on line length on the next line?